File: jput-core/src/main/java/cz/encircled/jput/unit/UnitPerformanceAnalyzer.java
package cz.encircled.jput.unit;

import cz.encircled.jput.model.MethodConfiguration;
import cz.encircled.jput.model.PerformanceTestRun;

import java.lang.reflect.Method;

/**
 * @author Vlad on 21-May-17.
 */
public interface UnitPerformanceAnalyzer {

    PerformanceTestRun buildRun(MethodConfiguration configuration, Method method);

    void addRun(PerformanceTestRun run, long elapsedTime);

    UnitPerformanceResult analyzeUnitTrend(PerformanceTestRun run, MethodConfiguration conf);

    String buildErrorMessage(UnitPerformanceResult result, MethodConfiguration conf);

}

File: jput-core/src/main/java/cz/encircled/jput/unit/UnitPerformanceAnalyzer.kt
package cz.encircled.jput.unit

import cz.encircled.jput.model.PerfTestConfiguration
import cz.encircled.jput.model.PerfTestExecution
import cz.encircled.jput.model.PerfTestResult
import java.lang.reflect.Method

/**
 * @author Vlad on 21-May-17.
 */
interface UnitPerformanceAnalyzer {

    fun buildTestExecution(configuration: PerfTestConfiguration, method: Method): PerfTestExecution

    fun analyzeUnitTrend(execution: PerfTestExecution, conf: PerfTestConfiguration): PerfTestResult

}
